File: MiniDOMAsteriskLib/CallManagers/Action/MailboxCount.vb
Imports System.Net.Sockets
Imports System.Threading
Imports System.Net
Imports minidom.CallManagers.Responses

Namespace CallManagers.Actions

    ''' <summary>
    ''' Check Mailbox Message Count
    ''' </summary>
    ''' <remarks></remarks>
    Public Class MailboxCount
        Inherits Action

        Private m_Mailbox As String
        Private m_ActionID As Nullable(Of Integer)
        
        Public Sub New()
            MyBase.New("MailboxCount")
        End Sub

        Public Sub New(ByVal mailbox As String)
            Me.New()
            Me.m_Mailbox = mailbox
        End Sub

        Public Sub New(ByVal mailbox As String, ByVal actionID As Integer)
            Me.New()
            Me.m_Mailbox = mailbox
            Me.m_ActionID = actionID
        End Sub

        Public ReadOnly Property Mailbox As String
            Get
                Return Me.m_Mailbox
            End Get
        End Property

        Public ReadOnly Property ActionID As Nullable(Of Integer)
            Get
                Return Me.m_ActionID
            End Get
        End Property

        Protected Overrides Function GetCommandText() As String
            If (Me.m_ActionID.HasValue) Then
                Return MyBase.GetCommandText() & "Mailbox: " & Me.m_Mailbox & vbCrLf & "Actionid: " & Me.m_ActionID
            Else
                Return MyBase.GetCommandText() & "Mailbox: " & Me.m_Mailbox & vbCrLf
            End If
        End Function
         

        Protected Overrides Function AllocateResponse(rows() As String) As ActionResponse
            Return New MailboxCountResponse(Me)
        End Function

        Public Overrides ReadOnly Property RequiredPrivilages As ActionPrivilageFlag
            Get
                Return ActionPrivilageFlag.call Or ActionPrivilageFlag.all
            End Get
        End Property

    End Class

End Namespace

File: MiniDOMAsteriskLib/CallManagers/Action/MailboxCount.cs
using DMD;
using minidom.CallManagers.Responses;

namespace minidom.CallManagers.Actions
{

    /// <summary>
    /// Check Mailbox Message Count
    /// </summary>
    /// <remarks></remarks>
    public class MailboxCount : Action
    {
        private string m_Mailbox;
        private int? m_ActionID;

        public MailboxCount() : base("MailboxCount")
        {
        }

        public MailboxCount(string mailbox) : this()
        {
            m_Mailbox = mailbox;
        }

        public MailboxCount(string mailbox, int actionID) : this()
        {
            m_Mailbox = mailbox;
            m_ActionID = actionID;
        }

        public string Mailbox
        {
            get
            {
                return m_Mailbox;
            }
        }

        public int? ActionID
        {
            get
            {
                return m_ActionID;
            }
        }

        protected override string GetCommandText()
        {
            if (m_ActionID.HasValue)
            {
                return base.GetCommandText() + "Mailbox: " + m_Mailbox + DMD.Strings.vbCrLf + "Actionid: " + m_ActionID;
            }
            else
            {
                return base.GetCommandText() + "Mailbox: " + m_Mailbox + DMD.Strings.vbCrLf;
            }
        }

        protected override ActionResponse AllocateResponse(string[] rows)
        {
            return new MailboxCountResponse(this);
        }

        public override ActionPrivilageFlag RequiredPrivilages
        {
            get
            {
                return ActionPrivilageFlag.call | ActionPrivilageFlag.all;
            }
        }
    }
}